bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TypeConverter.java

```diff
@@ -262,6 +331,57 @@ static boolean isConvertibleToFloatingPointTypes(Object value) {
         return convertibleTypes;
     }
 
+    private static boolean isConvertibleToRecordType_bstring(Object sourceValue, BRecordType targetType,
+                                                             List<TypeValuePair> unresolvedValues) {
+        if (!(sourceValue instanceof MapValueImpl)) {
+            return false;
+        }
+
+        TypeValuePair typeValuePair = new TypeValuePair(sourceValue, targetType);
+        if (unresolvedValues.contains(typeValuePair)) {
+            return true;
+        }
+        unresolvedValues.add(typeValuePair);
+
+        Map<String, BType> targetFieldTypes = new HashMap<>();
+        BType restFieldType = targetType.restFieldType;
+
+        for (BField field : targetType.getFields().values()) {
+            targetFieldTypes.put(field.getFieldName(), field.type);
+        }
+
+        MapValueImpl sourceMapValueImpl = (MapValueImpl) sourceValue;
+        for (Map.Entry targetTypeEntry : targetFieldTypes.entrySet()) {
+            String fieldName = targetTypeEntry.getKey().toString();
+
+            if (!sourceMapValueImpl.containsKey(StringUtils.fromString(fieldName))) {
+                BField targetField = targetType.getFields().get(fieldName);
+                if (Flags.isFlagOn(targetField.flags, Flags.REQUIRED)) {
+                    return false;
+                }
+            }
+        }
+
+        for (Object object : sourceMapValueImpl.entrySet()) {
+            Map.Entry valueEntry = (Map.Entry) object;
+            String fieldName = valueEntry.getKey().toString();
+
+            if (targetFieldTypes.containsKey(fieldName)) {
+                if (getConvertibleTypes(valueEntry.getValue(), targetFieldTypes.get(fieldName),
+                                        unresolvedValues).size() != 1) {
+                    return false;
+                }
+            } else if (!targetType.sealed) {
+                if (getConvertibleTypes(valueEntry.getValue(), restFieldType, unresolvedValues).size() != 1) {
```

Comment:
these two conditions can be merged

Reply:
I don't think they can be merged because of the else statement for 
`if (!targetType.sealed) {`

Reply:
ack